cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -202,6 +202,17 @@
                (src2 XmmMem)
                (dst WritableXmm))
 
+       ;; XMM (scalar or vector) blend op. The mask is used to blend between
+       ;; src1 and src2. This differs from a use of `XmmRmR` as the mask is
+       ;; implicitly in register xmm0; this special case exists to allow us to
+       ;; communicate the constraint on the `mask` register to regalloc2.
+       (XmmRmRBlend
+         (op SseOpcode)
+         (src1 Xmm)
+         (src2 XmmMem)
+         (mask Xmm)
```

Comment:
Another option here would be to use an `Option<Xmm>` field on `XmmRmR`. Happy to rework with that approach if the additional constructor here seems like too much.